### PR TITLE
Readme updates

### PR DESCRIPTION
**Pull request by JoshLove-msft**

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/README.md [hunk 1]
```diff
@@ -65,27 +65,38 @@ For more information about these resources, see [What is Azure Service Bus?](htt
...
 
-- A **Service Bus session processor** is scoped to a particular session-enabled queue or subscription, and is created using the Service Bus client. The session processor is almost identical to the standard processor, with the difference being that session management operations are exposed which only apply to session-enabled entities.
+- A [Service Bus session processor](https://docs.microsoft.com/en-us/dotnet/api/azure.messaging.servicebus.servicebussessionprocessor?view=azure-dotnet-preview) is scoped to a particular session-enabled queue or subscription, and is created using the Service Bus client. The session processor is almost identical to the standard processor, with the difference being that session management operations are exposed which only apply to session-enabled entities.
 
 For more concepts and deeper discussion, see: [Service Bus Advanced Features](https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-messaging-overview#advanced-features).
 
 ## Examples
 
-### Sending and receiving a message
+* [Send and receive a message](#sending-and-receive-a-message)
```
Comment (ShivangiReja) [inline sdk/servicebus/Azure.Messaging.ServiceBus/README.md:88]: nit: (#sending-and-receive-a-message) -> (#send-and-receive-a-message)

File: sdk/servicebus/Azure.Messaging.ServiceBus/README.md [hunk 2]
```diff
@@ -65,27 +65,38 @@ For more information about these resources, see [What is Azure Service Bus?](htt
...
+* [Send and receive a batch of messages](#send-and-receive-a-batch-of-messages)
+* [Complete a message](#complete-a-message)
+* [Abandon a message](#abandon-a-message)
+* [Defer a message](#defer-a-message)
+* [Dead letter a message](#dead-letter-a-message)
+* [Using the processor](#using-the-processor)
+* [Authenticating with Azure.Identity](#authenticating-with-azure.identity)
```
Comment (ShivangiReja) [inline sdk/servicebus/Azure.Messaging.ServiceBus/README.md:95]: Authenticating with Azure.Identity -> Link is broken

File: sdk/servicebus/Azure.Messaging.ServiceBus/README.md [hunk 3]
```diff
@@ -180,6 +185,8 @@ await receiver.CompleteAsync(receivedMessage);
 
 ### Abandon a message
 
+Abandoning a message releases our receiver's lock, which allows the message to be received by this or other receivers.
```
Comment (ShivangiReja) [inline sdk/servicebus/Azure.Messaging.ServiceBus/README.md:188]: Can we add some details for `Defer a message` as well?

File: sdk/servicebus/Azure.Messaging.ServiceBus/CHANGELOG.md [hunk 4]
```diff
@@ -1,7 +1,32 @@
...
+- Various performance improvements  
+  _(A community contribution, courtesy of [danielmarbach](https://github.com/danielmarbach))_
+- Improve the way exception stack traces are captured  
+  _(A community contribution, courtesy of [danielmarbach](https://github.com/danielmarbach))_
+  
+### Breaking Changes
+- Change from using a static factory method for creating a sendable message from a received message to instead
+  using a constructor  
+  _(A community contribution, courtesy of [danielmarbach](https://github.com/danielmarbach))_
+- CreateSessionProcessor method signature changed to accept a params array of session Ids instead of a single string.
```
[the remark this answers is not in the document]
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/CHANGELOG.md:25]: It's not an API break, but I was thinking it was still a binary break. On closer inspection though, you are right!
https://github.com/axelheer/dotnet-corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md#signatures

File: sdk/servicebus/Azure.Messaging.ServiceBus/MigrationGuide.md [hunk 5]
```diff
@@ -202,34 +202,22 @@ await receiver.CloseAsync();
...
-
-// the received message is a different type as it contains some service set properties
-IList<ServiceBusReceivedMessage> receivedMessages = await receiver.ReceiveBatchAsync(maxMessages: 2);
-
-foreach (ServiceBusReceivedMessage receivedMessage in receivedMessages)
-{
-    // get the message body as a string
-    string body = Encoding.Default.GetString(receivedMessage.Body.ToArray());
-    Console.WriteLine(body);
-}
```
[the remark this answers is not in the document]
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/MigrationGuide.md:232]: Good catch, this was due to renaming the snippet.

File: sdk/servicebus/Azure.Messaging.ServiceBus/CHANGELOG.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (ShivangiReja) [reply sdk/servicebus/Azure.Messaging.ServiceBus/CHANGELOG.md:25]: Ohh, yes we also change the name of the parameter.

File: sdk/servicebus/Azure.Messaging.ServiceBus/MigrationGuide.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/MigrationGuide.md:232]: Actually, both snippets are there. I will update the text though to mention that there are two ways to do this.

[the remark this answers is not in the document]
Comment (ShivangiReja) [reply sdk/servicebus/Azure.Messaging.ServiceBus/MigrationGuide.md:232]: I don't see receive batch example for v7 here - https://github.com/JoshLove-msft/azure-sdk-for-net/blob/preview2-readme/sdk/servicebus/Azure.Messaging.ServiceBus/MigrationGuide.md#sending-and-receiving-a-batch-of-messages.

File: sdk/servicebus/Azure.Messaging.ServiceBus/MigrationGuide.md [hunk 6]
```diff
@@ -200,24 +200,36 @@ await receiver.CloseAsync();
...
 
 // create the sender
 ServiceBusSender sender = client.CreateSender(queueName);
+```
+```C# Snippet:ServiceBusSendAndReceiveBatch
+IList<ServiceBusMessage> messages = new List<ServiceBusMessage>();
+messages.Add(new ServiceBusMessage(Encoding.UTF8.GetBytes("First")));
+messages.Add(new ServiceBusMessage(Encoding.UTF8.GetBytes("Second")));
+// send the messages
+await sender.SendAsync(messages);
```
Comment (ShivangiReja) [inline sdk/servicebus/Azure.Messaging.ServiceBus/MigrationGuide.md:217]: Can we update the description [here](https://github.com/JoshLove-msft/azure-sdk-for-net/blob/preview2-readme/sdk/servicebus/Azure.Messaging.ServiceBus/MigrationGuide.md#sending-and-receiving-a-batch-of-messages) and talk about SendAsync(IEnumerable<ServiceBusMessage>) api?

**Review by ShivangiReja — APPROVED** [review]

**Pull request merged**